From a940f2ca64e8b505ae149bd189bddb3d68054020 Mon Sep 17 00:00:00 2001
Message-Id: <a940f2ca64e8b505ae149bd189bddb3d68054020.1426052288.git.chang-joon.lee@intel.com>
In-Reply-To: <99f2c84a457bac958678aa60ef795f22948aa382.1426052288.git.chang-joon.lee@intel.com>
References: <99f2c84a457bac958678aa60ef795f22948aa382.1426052288.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Tue, 3 Mar 2015 19:07:09 +0530
Subject: [PATCH 2/5] SQUASHME! [VPG]: drm/i915: Remove pin & unpin fence for
 gen8+

Since a fence is not required for gen8+ there is no need to
pin & unpin a fence for the object.

Squash patch to :
commit 9a22fc4cb210b65a3ad2c7abdf8b14570e7c66dd
Author: Deepak S <deepak.s@intel.com>
Date:   Fri Jan 30 12:36:09 2015 +0530

    FOR_UPSTREAM [VPG]: drm/i915: Framebuffers need not be limited to
    256MB on gen8+.

Issue: GMINL-6718
Change-Id: I2aea7772586daacd5a08ef157621c638aea27a57
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 13285f2..22fada9 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2398,9 +2398,10 @@ intel_pin_and_fence_fb_obj(struct drm_device *dev,
 		ret = i915_gem_object_get_fence(obj);
 		if (ret)
 			goto err_unpin;
+
+		i915_gem_object_pin_fence(obj);
 	}
 
-	i915_gem_object_pin_fence(obj);
 	drm_gem_object_reference(&obj->base);
 
 	dev_priv->mm.interruptible = true;
@@ -2415,7 +2416,9 @@ err_interruptible:
 
 void intel_unpin_fb_obj(struct drm_i915_gem_object *obj)
 {
-	i915_gem_object_unpin_fence(obj);
+	struct drm_i915_private *dev_priv = obj->base.dev->dev_private;
+	if ((INTEL_INFO(dev_priv->dev)->gen <= 7))
+		i915_gem_object_unpin_fence(obj);
 	i915_gem_object_unpin_from_display_plane(obj);
 	drm_gem_object_unreference(&obj->base);
 }
-- 
1.7.9.5

